IPython/core/interactiveshell.py

```diff
@@ -1450,7 +1450,10 @@ def reset(self, new_session=True):
         # GUI or web frontend
         if os.name == 'posix':
             for cmd in ('clear', 'more', 'less', 'man'):
-                self.alias_manager.soft_define_alias(cmd, cmd)
+                try:
+                    name = self.magics_manager.magics['line'][cmd]
+                except KeyError:
+                    self.alias_manager.soft_define_alias(cmd, cmd)
```

Comment:
Do you have a reasoning to use `try:...Except KeyError` instead of checking:

```
if cmd not in self.magics_manager.magics['line']:
    self.alias_manager.soft_define_alias(cmd, cmd)
```

I would tend to have a preference for the `if ...` version , as `try/except` could swallow bugs in `soft_define_alias`, but I could easily be convinced.

Reply:
I agree with @Carreau here: I think it's better to use the `if`version and not the `try/except` one here.